### ISSUE TITLE
PaGMO Issues with Tudat

### ISSUE THREAD

**Issue by RajathPai95** [Tudat/tudat#583]
PaGMO example files not working, with path and namespace errors 

Paths like this are not finding files 
#include "pagmo/algorithms/de1220.hpp" 

Compile Output with errors attached .
[PagMO Example Cmpile Output.txt](https://github.com/Tudat/tudat/files/3836412/PagMO.Example.Cmpile.Output.txt)

None of the example applications also work. 
System Info : Windows 10 64 bit

**Comment by DominicDirkx** [comment]
Hi Rajath,

Can you confirm that this happens with a clean build of the tudat bundle, as pulled from Github?

Dominic

**Comment by RajathPai95** [comment]
Hey Dominic,

This is the same build we worked on last week. It was aspulled from Github. I'm not sure what the clean build refers to.

regards,
Rajath

**Comment by DominicDirkx** [comment]
HI Rajath,

Let me rephrase: are there any modifications in the code that you made, or is it the same as what was pulled from Github? 

Dominic

**Comment by RajathPai95** [comment]
Hey Dominic,

The modifications perhaps can be in my code, but I ran all the PaGMO Examples and still foudn them to be doing the same. THey don't recognise any of the PaGMo header files and the namespace pagmo also gives an error " expected namespace name".  I'm looking into such errors, but I fear this might be a bigger problem to understand for me.

Regards,
Rajath

**Comment by RajathPai95** [comment]
Hey Dominic,

Sorry for the confusing terminology. I tried to build all the Pagmo examples. (what i meant by running all examples)
No modifications were made to the tudatBundle, so the examples are untouched (from the git pull) and don't compile. 
I hope to have clarified my message. 

Regards,
Rajath

**Comment by DominicDirkx** [comment]
Hi Rajath,

Can you confirm that Tudat build correctly when USE_PAGMO was set to off? You mention some error outputs with " expected namespace name". Please provide the full error messages.

The error seems to be that it cannot find Pagmo, and certain Pagmo files. Can you verify that these files are present in tudatBundle, or not? Also, please attach your CMake output,

Best,

Dominic

**Comment by RajathPai95** [comment]
I cannot comment on the correctness of the build. How do i check for it?
Can you explain what that means? 

USE_PAGMO has been set to off in the initial build last week. I did switch it on, but it seems to have switched off again. Qt keeps forgetting some settings. I've attached the Cmake Output file . Hope it helps

The files are in pagmo2 folder and most of them exist. I found it surprising that the compiler failed to find it whe the path and the files both existed. 



[CMakeOutput.log](https://github.com/Tudat/tudat/files/3840236/CMakeOutput.log)

**Comment by DominicDirkx** [comment]
The correctness of the build can be checked by compiling the code and running the unit test. 

How did you switch on 'USE_PAGMO'? What do you mean bu forgetting 'some settings': which ones, and why do you come to this conclusion?

What do you mean by 'most of them exist'? Which ones don't?

**Comment by DominicDirkx** [comment]
Concerning your CMake output, as in previous posts, I am referring to the output in 'General Messages', not the file you have attached here.

**Comment by RajathPai95** [comment]
Hey Dominic,

To answer all you questions,

The last time i checked, all the unit test ran fine(tudat)

I switched on the use_Pagmo in Qt build settings. The Compile output shows PaGMO disabled now when i had a look. Should I switch it on from the CmakeLists?

By forgetting some settings, I mean Build settings. Using current executable,  Qt would run different applications than what I would want it to run, and what I was setting it up (sometimes a previously run example application).

Some of the header files dont exist within PaGMO which are possibly in the code from MIranda's thesis. One of them was 'serialization.hpp'. Perhaps they're non existrent in pagmo2, but were part of old PaGMO.

Compile Output as asked. 
[compile output 1311.txt](https://github.com/Tudat/tudat/files/3841291/compile.output.1311.txt)
'

**Comment by DominicDirkx** [comment]
Hi Rajath,

Just to make sure, to turn USE_PAGMO to on, toggle USE_PAGMO to ON under Project->Build->Cmake, and then don't forget to 'Apply Configuration Changes'. You do not need to change anything in the CMakeLists. 

The structure of the new pagmo is quite different from the old one, so it may well be possible that some of the old include statements are no longer present.

Looking back to your compile output in your first post, there do seem to be some discrepancies w.r.t. what I get on my installation. If you can drop by at 11:30, I can have a better look, this will be more expedient for this error that through Github,

Dominic

**Comment by djrobertjin** [comment]
I also experience issues  compiling when I check USE_PAGMO on my mac:

`[100%] Building CXX object tudatExampleApplications/libraryExamples/PaGMOEx/CMakeFiles/application_PagmoCEC2013Comparison.dir/cec2013OptimizerComparison.cpp.o
In file included from /Users/drj/Documents/NumericalAstrodynamics/tudatBundle/tudatExampleApplications/libraryExamples/PaGMOEx/cec2013OptimizerComparison.cpp:5:
/Users/drj/Documents/NumericalAstrodynamics/tudatBundle/pagmo2/include/pagmo/problems/cec2013.hpp:36:2: error: The cec2013.hpp header was included but the CEC 2013 problem is not supported on the current platform.
#error The cec2013.hpp header was included but the CEC 2013 problem is not supported on the current platform.
 ^
1 error generated.
make[2]: *** [tudatExampleApplications/libraryExamples/PaGMOEx/CMakeFiles/application_PagmoCEC2013Comparison.dir/cec2013OptimizerComparison.cpp.o] Error 1
make[1]: *** [tudatExampleApplications/libraryExamples/PaGMOEx/CMakeFiles/application_PagmoCEC2013Comparison.dir/all] Error 2
make: *** [all] Error 2
21:53:58: The process "/Applications/CMake.app/Contents/bin/cmake" exited with code 2.
Error while building/deploying project TudatBundle (kit: Desktop (x86-darwin-generic-mach_o-32bit))
When executing step "CMake Build"
21:53:58: Elapsed time: 21:03.
`

I turned off PAGMO and it compiled fine.

**Comment by DominicDirkx** [comment]
Hi,

Did you encounter this issue with a recent (e.g. last 1-2 weeks) download of Tudat, or an older one?

Best,

Dominic

**Comment by djrobertjin** [comment]
I encountered it on Tuesday.

**Comment by DominicDirkx** [comment]
Could you let me know which compiler you are using?

**Comment by djrobertjin** [comment]
I'm using XCode command line tools

**Comment by DominicDirkx** [comment]
Thanks, apparently there is a small Mac incompatibility with using one small aspect of Pagmo. We'll leave this issue open so that we're reminded to correct it,

Dominic
